pkg/controller/nstemplateset/nstemplateset_controller_test.go

```diff
@@ -1912,8 +1939,8 @@ func getTemplateContent(decoder runtime.Decoder) func(tierName, typeName string)
 			}
 		case "basic":
 			switch typeName {
-			case ClusterResources: // assume that this tier has no "cluster resources" template
-				return nil, nil
+			case ClusterResources:
```

Comment:
We explicitly made ClusterResource optional: https://github.com/codeready-toolchain/api/blob/master/pkg/apis/toolchain/v1alpha1/nstemplatetier_types.go#L22
So, we needed a test which would make sure a tier without ClusterResource would work.

But I think we can actually make ClusterResource required but then let's fix the API.

Reply:
ah, I totally forgot that it's optional.

> But I think we can actually make ClusterResource required but then let's fix the API.

sure, but until it's required, we should test also the optionality - let me fix that

Reply:
done 5129524
I had to change the logic for creating the ClusterResourceQuotas resource during the test setup - we need to parse the quota from yaml file, otherwise, the object contains slightly different formating that the one retrieved from the template - this causes that it looks like the resource was modified, thus generation incremented